arch/Kconfig

```diff
@@ -846,6 +846,9 @@ config BOARD
 	  if not found we look for the linker file in
 	  soc/<arch>/<family>/<series>
 
+config BOARD_REVISION
+	def_string "$BOARD_REVISION"
```

Comment:
would be nice to add `help` as well.

by the way, is `arch/Kconfig` really the right place to add this config? IMO, `boards/Kconfig` would be more suited for this purpose.

Reply:
Agreed, `arch/Kconfig` is not the right place for this. +1 for `boards/Kconfig`.

Reply:
This is being added right under the definition for CONFIG_BOARD. Are you saying we should move both of them to boards/? Moving just this one would not make sense. 

@nashif @stephanosio